p2p/exchange.go

```diff
@@ -157,8 +166,20 @@ func (ex *Exchange[H]) Head(ctx context.Context, opts ...header.HeadOption[H]) (
 	)
 	for _, from := range peers {
 		go func(from peer.ID) {
+			span.AddEvent("sending request to peer",
```

Comment:
```suggestion
			span.AddEvent("requesting peer",
```

Reply:
Actually, every peer request better be a span so we can see the time it took for each peer.

Reply:
In general, if your span produces lots of finish/start events, it's asking for another span

Reply:
Also, instead of passing peerID to every event, we can use [baggage](https://opentelemetry.io/docs/concepts/signals/baggage/), which is like an attribute that propagates to child spans. Together with spans this will decrease load and make things more readable